scenes/scene_base.gd

```diff
@@ -0,0 +1,26 @@
+extends XRToolsSceneBase
+
+func _ready():
+	super._ready()
+
+	var webxr_interface = XRServer.find_interface("WebXR")
+	if webxr_interface:
+		XRToolsUserSettings.webxr_primary_changed.connect(self._on_webxr_primary_changed)
```

Comment:
Shouldn't this call `_on_webxr_primary_changed` atleast once to setup the right action first time the scene is loaded?

Reply:
I would also change `if webxr_interface` to `if webxr_interface and webxr_interface.is_initialised()`, the webXR interface will always be there, even if it is not being used. in its present form this code would also run if OpenXR is used.

Reply:
> Shouldn't this call _on_webxr_primary_changed atleast once to setup the right action first time the scene is loaded?

It does that! It's the next line after the ones you quoted above. :-)

> I would also change `if webxr_interface` to `if webxr_interface and webxr_interface.is_initialised()`, the webXR interface will always be there, even if it is not being used.

It's normal to enter and exit WebXR, sometimes multiple times. I think this really does need to be setup even if WebXR is just available and not initialized yet, because you want it to be ready for when WebXR is entered.

In fact, checking if WebXR is initialized here will break the demo, because when the web page is first loaded, you're not in WebXR mode, but the staging scene is already running.

>  in its present form this code would also run if OpenXR is used.

No, OpenXR and WebXR actually can't co-exist in the same build. WebXR will _only_ be compiled in a web build, and OpenXR will _never_ be compiled in a web build.

Another way to think of it, is that all the checks to see if WebXR is available are also implicitly checking if we are in a web build.

Reply:
ah offcourse, I keep forgetting webXR is very specific to the html platform and only available there.